code/public.tf

```diff
@@ -20,146 +20,112 @@ locals {
   http_to_https_redirect_rule     = format("%s-appgw-http-to-https-redirect-rule", local.project)
 }
 
-resource "azurerm_application_gateway" "api_gateway" {
-  name                = format("%s-api-gateway", local.project)
-  resource_group_name = azurerm_resource_group.rg_public.name
-  location            = azurerm_resource_group.rg_public.location
-
-  sku {
-    name = "WAF_v2"
-    tier = "WAF_v2"
-  }
-
-  enable_http2 = true
-
-  ssl_policy {
-    policy_type = "Predefined"
-    policy_name = "AppGwSslPolicy20170401S"
-  }
-
-  gateway_ip_configuration {
-    name      = format("%s-appgw-gw-ip-configuration", local.project)
-    subnet_id = module.subnet_public.id
-  }
-
-  frontend_port {
-    name = local.frontend_http_port_name
-    port = 80
-  }
-
-  frontend_port {
-    name = local.frontend_https_port_name
-    port = 443
-  }
+data "azurerm_key_vault_certificate" "app_gw_platform" {
+  name         = var.app_gateway_certificate_name
+  key_vault_id = azurerm_key_vault.key_vault.id
+}
 
-  frontend_ip_configuration {
-    name                 = local.frontend_ip_configuration_name
-    public_ip_address_id = azurerm_public_ip.apigateway_public_ip.id
-  }
 
-  backend_address_pool {
-    name         = local.backend_address_pool_name
-    fqdns        = [trim(azurerm_private_dns_a_record.private_dns_a_record_api.fqdn, ".")]
-    ip_addresses = []
-  }
+# Application gateway: Multilistener configuraiton
+module "app_gw" {
+  source = "git::https://github.com/pagopa/azurerm.git//app_gateway?ref=v2.0.2"
 
-  backend_http_settings {
-    name                  = local.http_setting_name
-    host_name             = trim(azurerm_private_dns_a_record.private_dns_a_record_api.fqdn, ".")
-    cookie_based_affinity = "Disabled"
-    path                  = ""
-    port                  = 80
-    protocol              = "Http"
-    request_timeout       = 60
-    probe_name            = "probe-apim"
-  }
+  resource_group_name = azurerm_resource_group.rg_public.name
+  location            = azurerm_resource_group.rg_public.location
+  name                = format("%s-api-gateway", local.project)
 
-  probe {
-    host                                      = trim(azurerm_private_dns_a_record.private_dns_a_record_api.fqdn, ".")
-    minimum_servers                           = 0
-    name                                      = "probe-apim"
-    path                                      = "/status-0123456789abcdef"
-    pick_host_name_from_backend_http_settings = false
-    protocol                                  = "Http"
-    timeout                                   = 30
-    interval                                  = 30
-    unhealthy_threshold                       = 3
-
-    match {
-      status_code = ["200-399"]
+  # SKU
+  sku_name = "WAF_v2"
+  sku_tier = "WAF_v2"
```

Comment:
probably here we need a variable to use standard_v2 sku in uat and WAF_v2 in prod.
Also in uat we need to explicit disable waf